src/main/java/gov/nasa/jpl/mbee/mdk/generator/DocumentGenerator.java

```diff
@@ -277,11 +277,20 @@ public Section parseView(Element view) {
                     //if (expose.size() == 1 && expose.get(0) instanceof Diagram) {
                     for (Element ex : elementImports) {
                         if (ex instanceof Diagram) {
-                            Image image = new Image();
-                            List<Object> images = new ArrayList<Object>();
-                            images.add(ex);
-                            image.setTargets(images);
-                            viewSection.addElement(image);
+                            String diagramType = Application.getInstance().getProject().getDiagram((Diagram) ex).getDiagramType().getType();
+                            if(diagramType.equals("Generic Table") ||diagramType.equals("Instance Table") || diagramType.equals("Verify Requirement Matrix") || diagramType.equals("SysML Allocation Matrix") || diagramType.equals("Satisfy Requirement Matrix")){
```

Comment:
Yes, i saw that they walk up the hierarchy. Agreed on adding only the necessary ones. Ill check.